core/modules/wbase/Task.h

```diff
@@ -68,19 +74,38 @@ class TaskScheduler {
     virtual void taskCancelled(Task*)=0;///< Repeated calls must be harmless.
 };
 
+/// Used to find tasks that are in process for debugging with Task::tSeq.
+/// This is largely meant to track down incomplete tasks in a possible intermittent
+/// failure and should probably be removed when it is no longer needed.
+/// It depends on code in BlendScheduler to work. If the decision is made to keep it
+/// forever, dependency on BlendScheduler needs to be re-worked.
+struct IdSet {
+    void add(int id) { std::lock_guard<std::mutex> lock(mx); _ids.insert(id); }
+    void remove(int id) { std::lock_guard<std::mutex> lock(mx); _ids.erase(id); }
+    std::string toString() {
+        std::ostringstream os;
+        std::lock_guard<std::mutex> lock(mx);
+        os << "count=" << _ids.size() << " ";
+        for(auto j: _ids) {
+            os << j << ", ";
```

Comment:
There is `utils::printable()` in `utils/IterableFormatter.h` which does nice formatting of sequences and maps. I personally only start sweating when I see `toString()` method, this is so Java and I don't speak Java well :smirk_cat: